### ISSUE TITLE
epw file with holidays sometimes terminates with a "** Severe ** bad allocation" error

### ISSUE THREAD

**Issue by yuizi** [NREL/EnergyPlus#9743]
Issue overview
--------------
I am running e+9.5 with exactly the same IDF, but sometimes I get errors.

I run e+ 9.5 on the command line as follows
```
> energyplus -x -w weather.epw Input.idf
```
Why does the error occur even if I run e+ under the same conditions?
I don't know the cause of the error at all, does anyone know what the cause is?

The files used for the calculations are attached.
[mydata.zip](https://github.com/NREL/EnergyPlus/files/10031399/mydata.zip)

<img width="985" alt="capture" src="https://user-images.githubusercontent.com/4317135/202444558-1e1eebb3-6deb-4a91-857e-cfdc7e3ab48a.png">

**Comment by mjwitte** [comment]
@yuizi Running in the debugger reveals that EnergyPlus is throwing a vector subscript error while processing the holidays in the epw file header. This type of error can produce unpredictable results when running the release exe.

The workaround for this is to remove the holidays from the epw header (or simply set the number of holidays to zero) and then  use `RunPeriodControl:SpecialDays` to define the holidays in the idf file. This will be fixed for v23.1.

```
HOLIDAYS/DAYLIGHT SAVINGS,No,0,0,14,New Year's day,January 1,Coming-of-Age Day,2nd Monday in January,National Foundation Day,February 11,Vernal Equinox Day,March 21,Greenery Day,April 29,Constitution Day,May 3,National holiday,May 4,Chirden's Day,May 5,Respect-for-the-Aged Day,3rd Monday in September,Autumnal Equinox Day,September 23,Health and Sports Day,2nd Monday in October,Culture Day,November 3,LaborThanksgiving Day,November 23,The Emperor's Birthday,December 23

HOLIDAYS/DAYLIGHT SAVINGS,No,0,0,0
```

**Comment by yuizi** [comment]
@mjwitte Thank you for investigating the cause of the error and providing the workaround.
I have been having trouble finding the cause of the error for the past few days.
Thank you so much for your help.
